lib/stud/task.rb

```diff
@@ -36,5 +36,14 @@ def stop!
     def self.interrupted?
       Thread.current[:stud_task_interrupted]
```

Comment:
I agree with you @ph and @colinsurprenant , interruption is completely broken. Test are not catching this interruption and only checking for the counter.

Not an expert here, but my Thread.current is not catching the right thread in my experiments. We should point to the exact thread we aim to close I guess.